### ISSUE TITLE
`ttnn.pad`: HEIGHT_SHARDED + width front-padding + sharded output produces wrong results

### ISSUE THREAD

**Issue by VirdhatchaniKN** [tenstorrent/tt-metal#49492]
### Description

When `ttnn.pad` is called with a HEIGHT_SHARDED RM input, width front-padding (non-zero `before` on the W axis), and a HEIGHT_SHARDED output, the result is garbage (PCC ~0.002 vs torch reference).

The bug is in the `PadRmShardedWidthOnlyProgramFactory` which miscomputes front-pad offsets. This factory is invoked when `pad.cpp` detects HEIGHT_SHARDED input with width padding and a non-interleaved output config — it enters the HEIGHT_SHARDED-specific width-pad decomposition path.

**Note:** HEIGHT_SHARDED + width front-pad + DRAM (interleaved) output works correctly (PCC 1.0) because it bypasses the sharded factory and uses the default `PadRmReaderWriterMultiCoreDefaultProgramFactory`.

### Reproducer

```python
import torch
import ttnn

device = ttnn.open_device(device_id=0)

input_shape = [1, 1, 64, 128]
padding_spec = [(0, 0), (0, 0), (0, 0), (64, 0)]  # front-pad W by 64

torch.manual_seed(0)
torch_input = torch.randn(input_shape, dtype=torch.bfloat16)
torch_output = torch.nn.functional.pad(torch_input, [64, 0, 0, 0], value=0.0)

# Create HEIGHT_SHARDED input
num_cores = device.compute_with_storage_grid_size()
total_cores = min(num_cores.x * num_cores.y, 64)
shard_h = (64 + total_cores - 1) // total_cores
grid = ttnn.CoreRangeSet([ttnn.CoreRange(
    ttnn.CoreCoord(0, 0),
    ttnn.CoreCoord(min(total_cores, num_cores.x) - 1, (total_cores - 1) // num_cores.x)
)])
in_cfg = ttnn.MemoryConfig(
    ttnn.TensorMemoryLayout.HEIGHT_SHARDED, ttnn.BufferType.L1,
    ttnn.ShardSpec(grid, (shard_h, 128), ttnn.ShardOrientation.ROW_MAJOR)
)
out_cfg = ttnn.MemoryConfig(
    ttnn.TensorMemoryLayout.HEIGHT_SHARDED, ttnn.BufferType.L1,
    ttnn.ShardSpec(grid, (shard_h, 192), ttnn.ShardOrientation.ROW_MAJOR)
)

tt_input = ttnn.from_torch(torch_input, layout=ttnn.ROW_MAJOR_LAYOUT,
                           dtype=ttnn.bfloat16, device=device, memory_config=in_cfg)
tt_output = ttnn.pad(tt_input, padding_spec, value=0.0, memory_config=out_cfg)
actual = ttnn.to_torch(tt_output)

pcc = torch.corrcoef(torch.stack([torch_output.float().flatten(),
                                   actual.float().flatten()]))[0, 1].item()
print(f"PCC: {pcc:.6f}")  # Prints ~0.002250

ttnn.close_device(device)
```

### Expected behavior
PCC should be >= 0.9999.

### Actual behavior
PCC is ~0.002 — output is effectively garbage.

### Root cause
`PadRmShardedWidthOnlyProgramFactory` does not correctly handle the `before` (front-pad) offset on the width dimension. The factory was designed for trailing-only width padding on HEIGHT_SHARDED tensors.
